apps/clock/js/alarm_database.js

```diff
@@ -83,12 +83,10 @@ define(function(require) {
 
       // Map '1111100' string bitmap to a repeat object with day properties.
       if (typeof alarm.repeat === 'string') {
-        var days = ['monday', 'tuesday', 'wednesday', 'thursday', 'friday',
-                    'saturday', 'sunday'];
         var newRepeat = {};
-        for (var i = 0; i < alarm.repeat.length && i < days.length; i++) {
-          if (alarm.repeat[i] === '1') {
-            newRepeat[days[i]] = true;
+        for (var i = 0; i < alarm.repeat.length; i++) {
+          if (alarm.repeat[i.toString()] === '1') {
+            newRepeat[i.toString()] = true;
```

Comment:
This part looks fine, but we'll also need to add a conversion from the current database format (i.e. `{ monday: true }` to `{ "1": true }` or whatever). Fortunately you can change the format at will as long as we update this function to perform the appropriate normalization.